### ISSUE TITLE
Can't enter complex sub-query during in(),out(),...

### ISSUE THREAD

**Issue by cgreenberg** [orientechnologies/orientdb#6866]
## OrientDB Version, operating system, or hardware.
- v2.0 SNAPSHOT[ ] - .18[ ] .17[ ] .16[ ] .15[ ] .14[ ] .13[ ] .12[ ] .11[ ] .10[ ] .9[ ] .8[ ] .7[ ] .6[ ] .5[ ] .4[ ] .3[ ] .2[ ] .1[ ] .0[ ]
- v2.1 SNAPSHOT[ ] - .16[ ] .15[ ] .14[ ] .13[ ] .12[ ] .11[X] .10[ ] .9[ ] .8[ ] .7[ ] .6[ ] .5[ ] .4[ ] .3[ ] .2[ ] .1[ ] .0[ ]
- v2.2 SNAPSHOT[ ] - .rc1[ ] .beta2[ ] .beta1[ ]

## Operating System
- [ ] Linux
- [X] MacOSX
- [ ] Windows
- [ ] Other Unix
- [ ] Other, name? 

## Expected behavior and actual behavior
I have heard you should be able to add conditions when calling `out()`, for example:
`select out('friendswith')[condition].out('friendswith')[condition] from E`
This should only select vertices that are connected with 'friends with' where a condition is applied. I can make this work for simple conditions like `type="best"` but I can't do, for example, `type IN ["best","second best"]`.

Also, I can't find the documentation on this conditioning anywhere.

**Comment by luigidellaquila** [comment]
Hi @cgreenberg 

Your query is checking the condition on the `vertex`.
Are you trying to evaluate a condition on the `edge`?
In this case, you have to use `outE()[condition].inV()`, eg. this will return my best friend:

```
select outE('friendswith')[type="best"].inV() from E where name = 'Luigi'
```

I hope it helps

Thanks

Luigi

**Comment by cgreenberg** [comment]
Hi, yes, you're right, the condition should be on `outE` not `out`. But I'm still having the same problem that I can only write `type="best"` and I can't write `type in ["best","worst"]`

**Comment by luigidellaquila** [comment]
Hi @cgreenberg 

Ok, got it, probably it's a parsing problem. I'll check it asap

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @cgreenberg 

I just tried it with 2.2.12 and it works fine for me. Which version are you using?

Thanks

Luigi

**Comment by cgreenberg** [comment]
@luigidellaquila I'm using 2.1.11. Was this fixed in recent versions?

**Comment by luigidellaquila** [comment]
Hi @cgreenberg 

Yes, we did a lot of fixes on square bracket filtering in v 2.2.x, if you have a chance to upgrade you'll have a lot of advantages

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Closing this, as it won't be fixed in 2.1.x series (fixes on that release only apply for paid contracts)

Thanks

Luigi
